### PR TITLE
http3-client: send multiple identical requests

### PR DESCRIPTION
**Pull request by LPardue**
Adds a new CLI option `-n`,`--requests` that allows the client to send multiple identical requests at the same time. Default is 1.

```
http3-client -n3 https://cloudflare-quic.com
```

The PR is is a lowest impact way to add this feature. The downsides are: only record and report timing of all requests, request bodies printed to stdout will get horribly intermingled.

### CODE REVIEW

File: examples/http3-client.rs [hunk 1]
```diff
@@ -46,6 +46,7 @@ Options:
   --no-verify              Don't verify server's certificate.
   --no-grease              Don't send GREASE.
   -H --header HEADER ...   Add a request header.
+  -n --number REQUESTS     Send the given number of identical requests. [default: 1]
```
Comment (ghedo) [inline examples/http3-client.rs]:
```suggestion
  -n --requests REQUESTS   Send the given number of identical requests [default: 1].
```

`--requests` seems like is a nicer name (same as h2load).

File: examples/http3-client.rs [hunk 2]
```diff
@@ -269,19 +273,27 @@ fn main() {
...
 
-                Err(e) => {
-                    error!("failed to send request {:?}", e);
-                    break;
-                },
-            };
+                let s = match h3_conn.send_request(
+                    &mut conn,
+                    &req,
+                    body.as_ref().is_none(),
```
Comment (ghedo) [inline examples/http3-client.rs]:
```suggestion
                    body.is_none(),
```

File: examples/http3-client.rs [hunk 3]
```diff
@@ -269,19 +273,27 @@ fn main() {
...
-                if let Err(e) = h3_conn.send_body(&mut conn, s, &body, true) {
-                    error!("failed to send request body {:?}", e);
-                    break;
+                    Err(e) => {
+                        error!("failed to send request {:?}", e);
+                        break;
+                    },
+                };
+
+                if let Some(body) = body.as_ref() {
```
Comment (ghedo) [inline examples/http3-client.rs]:
```suggestion
                if let Some(body) = &body {
```

File: examples/http3-client.rs [hunk 4]
```diff
@@ -269,19 +273,27 @@ fn main() {
...
+                    if let Err(e) = h3_conn.send_body(&mut conn, s, &body, true) {
```
Comment (ghedo) [inline examples/http3-client.rs]:
```suggestion
                    if let Err(e) = h3_conn.send_body(&mut conn, s, body, true) {
```

File: examples/http3-client.rs [hunk 5]
```diff
@@ -315,19 +327,26 @@ fn main() {
                     },
 
                     Ok((_stream_id, quiche::h3::Event::Finished)) => {
-                        info!(
-                            "response received in {:?}, closing...",
-                            req_start.elapsed()
-                        );
+                        complete_req += 1;
```
Comment (ghedo) [inline examples/http3-client.rs]:
```suggestion
                        complete_req += 1;

```

File: examples/http3-client.rs [hunk 6]
```diff
@@ -75,6 +76,11 @@ fn main() {
     // Request headers (can be multiple).
     let req_headers = args.get_vec("--header");
 
+    let num_req = args.get_str("--number");
```
Comment (ghedo) [inline examples/http3-client.rs]: `num_req` => `reqs_count` and `complete_req` => `reqs_complete`?

File: examples/http3-client.rs [hunk 7]
```diff
@@ -315,19 +327,26 @@ fn main() {
                     },
 
                     Ok((_stream_id, quiche::h3::Event::Finished)) => {
-                        info!(
-                            "response received in {:?}, closing...",
-                            req_start.elapsed()
-                        );
+                        complete_req += 1;
+                        debug!("{}/{} responses received", complete_req, num_req);
```
Comment (LPardue) [inline examples/http3-client.rs]: We could record time elapsed here. I don't have a strong opinion if we should

**Comment by LPardue** [comment]
I think all feedback is addressed now, PTAL

**Pull request merged**